### PR TITLE
STR-632: functional tests ability to use production env settings

### PR DESCRIPTION
**Pull request by storopoli**
## Description

- Adds the ability to use custom chains in `strata-reth`.
- Adds a global env `"devnet"` that uses the devnet chain. This allows new or existing tests to use the "production" environment that emulates Strata's devnet.
- Fixes the `fn_bridge_withdraw_happy` to use the pre-funded addresses in #467 which provides a nice speed up since we don't need to generate coinbases to these addresses.
- I had no idea why I had to make `fn_bridge_duties` more robust. Lots of wasted hours but managed to make it better.

### Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update
-   [ ] Refactor
-   [x] New or updated tests

## Checklist

-   [x] I have performed a self-review of my code.
-   [x] I have commented my code where necessary.
-   [x] I have updated the documentation if needed.
-   [x] My changes do not introduce new warnings.
-   [x] I have added tests that prove my changes are effective or that my feature works.
-   [x] New and existing tests pass with my changes.

## Related Issues

STR-632

### CODE REVIEW

**Review by bewakes — COMMENTED** [review]
Small query, othewise LGTM.

File: functional-tests/entry.py [hunk 1]
```diff
@@ -373,6 +377,8 @@ def main(argv):
         # Operator lag is a test that checks if the bridge can handle operator lag.
         # It is also useful for testing the reclaim path.
         "operator_lag": BasicEnvConfig(101, message_interval=10 * 60 * 1_000),
+        # Devnet production env
+        "devnet": BasicEnvConfig(101, custom_chain="devnet"),
```
[the remark this answers is not in the document]
Comment (bewakes) [reply functional-tests/entry.py:381]: Nevermind, devnet vs local development. 🤦

[the remark this answers is not in the document]
Comment (storopoli) [reply functional-tests/entry.py:381]: Yes

[the remark this answers is not in the document]
Comment (storopoli) [reply functional-tests/entry.py:381]: I didn't want to touch the json files in `strata-reth`.

**Pull request merged**